### ISSUE TITLE
Gestures for (re)capitalize letters and return flat theme 

### ISSUE THREAD

**Issue by brandmauer** [Julow/Unexpected-Keyboard#305]
1. All versions 1.2* have light "shadows" in dark theme. "Flat button look" in version 1.9* and your [screenshots ](https://github.com/Julow/Unexpected-Keyboard/blob/master/metadata/android/en-US/images/phoneScreenshots/2.png)- more better (imho)
Please return (or add) "flat theme" in new version. 
And maybe color adjustment for accented letters?

2. Please, add more power for gestures. For example: 

swipe up- capitalize letter
swipe down- small letter
swipe up right- letter from second layout (capital)
swipe up left- letter from second layout (small letter)
swipe up left- "Ctrl" mod for buttons (for "ctrl+c" "ctrl+v" "ctrl+z" "ctrl+a" and other actions  etc)
swipe left- move cursor to left
swipe right- move cursor to right

My gestures is only for example. I think -this tricks add more speed for keyboard.

**Comment by Julow** [comment]
I plan to add styling options in the future but that's not the highest priority.

That are great ideas but the layouts currently are pretty crowded. Would you be able to design a layout with some of these ideas plus all the symbols it already has ?

**Comment by brandmauer** [comment]
I`m a designer- not a developper. But if you give me sample or instruction- i can try )))
I think it`s about xml configuration file for buttons? 
Im ready to try to make english-russian "turbo config" ))

**Comment by Julow** [comment]
I have made a step in your direction with a6fe5ca. I wrote some explanation in qwerty.xml.

**Comment by brandmauer** [comment]
Wow! Exactly what is needed!

...I think you need add for this "settings to adjust color" (visibility) for secondary symbols. Otherwise- it will be "visual hell" )))

P.S. Can i make this turbo-layout in version  1.19* (with "Flat button look")?

**Comment by brandmauer** [comment]
Maybe my idea is bad and it is better to leave only a diagonal swipe, to avoid false key presses?
What do you think?

```
1 7 2       1    2
5 0 6   >   .  0 .
3 8 4       3 .  4
```

**Comment by Julow** [comment]
Spacing between keys on the third row doesn't look wrong. These keys are specified to be smaller with `width="0.96"`.

You are not using the `loc` prefix correctly. The `loc` modifier is for example to add French accents on the English layout. English users don't want to see them and French users like the English layout when programming.
In your case, you shouldn't use it.

The size of the symbols on the keyboard is currently defined globally, per symbol. It is controlled by flag `FLAG_SMALLER_FONT` in `KeyValue.java`.
The flag `FLAG_SECONDARY` can make a symbol dimmer but it is not possible to specify a color. You'd need to modify the rendering code (`Keyboard2View.java`).

> Maybe my idea is bad and it is better to leave only a diagonal swipe, to avoid false key presses?

Positions 5-8 are used for arrow keys, the space bar and others. Also, you asked for it.
If you think they increase false key presses, you can perhaps remove the capital letters ?

**Comment by brandmauer** [comment]
> In your case, you shouldn't use it.
1. Really need "invisible button" for layout- same as "`loc `" modifier. **How  to do it?**

> Positions 5-8 are used for arrow keys, the space bar and others
> 2. They still dont work ))) 

![IMG_20230306_121419](https://user-images.githubusercontent.com/25385530/223067765-9db24380-0f99-4dcd-a7f1-6d80f148caf1.jpg)

Other things are not important....

**Comment by Julow** [comment]
There's no concept of "invisible and working button", you need to code it.

What do you mean by they don't work ?

**Comment by brandmauer** [comment]
Need modificator (as "loc"), for hide letters label. See on photo above- for Uppercase letters.

At this time all letters with 'loc"- dont work, use 9 letters on one buttons- is visual madness )))

I think we have problems with the translation))

A much needed modifier (like "loc") to hide the label of the letters. (for example: see the photo above - for capital letters.)

At the moment, all letters with 'loc' - do not work. Using 9 letters on one button is "visual madness"! )))

**Comment by senselocke** [comment]
Gboard handles case-changing like this:
1. Select word(s) (up to ~50 characters)
2. Press "Shift" to cycle between:
  a) lowercase,
  b) Titlecase, and
  c) UPPERCASE

A similar implementation would add the functionally without adding visual clutter or more keys. If you do so, a larger string size would be appreciated. (BTW this is SUCH a cool keyboard!! The "Case Cycling" is one of the VERY few GBOARD features I miss).

Alternately, as a per-key option, you could add a setting to switch "hold for repeat" with "hold for Capital". Holding the 'a' key would produce 'A' instead of 'aaaaaa…'. This would also aboid extra clutter, and serve as a way to disable auto-repeat if desired.

**Comment by Julow** [comment]
@senselocke That's very interesting, thanks! The keyboard doesn't do any editing at the moment, which is a long standing issue https://github.com/Julow/Unexpected-Keyboard/issues/242

Is anyone interested in implementing the "hide label" or "hold for capital" ?

**Comment by Spike-from-NH** [comment]
I take @Julow's side. Erasing a word and RETYPING it in capitals is a "do-what-I-meant" function that conflicts with his stated philosophy (first word of program description is "lightweight"). Such editing belongs in the editor, not the keyboard.

In moving swipe assignments away from the edges of the keyboard (see my #354), I too had problems when a regular-sized key had both north and northwest swipes. The solution is not to use this option, not for Julow to remove it.

I second OP's desire for more control over appearance, though this is at a low priority. To really dream: Any keyboard, row, or key could have STYLE="…" (including gradients) like CSS. I have no interest in making the legends disappear!

One user's perfection is another user's annoyance. I think the existing ability to customize is the right solution. My personal layout already would seem "cluttered" to others—but with exactly the characters I want to type!

**Comment by Spike-from-NH** [comment]
PS—If redesign of the XML is on the table, using compass-points rather than 1 thru 8 eliminates some needless thinking. So—
 <key key="р" nw="H" ne="|" sw="h"  w="loc left">

But this is hardly crucial, and breaks existing files in this codebase and on many people's phones unless you support both versions.

**Comment by Julow** [comment]
@Spike-from-NH An editing mode might be challenging to implement in a lightweight way but it's still possible.

I am not against more advanced styling but I don't like the idea of specifying the style in the layouts. The themes and the layouts are currently separated and I would like to keep it that way. However, styling keys differently depending on what are on them or a "class" system like in HTML might be interesting.

> using compass-points rather than 1 thru 8 eliminates some needless thinking.

I like this idea. These could be added as alias to the current key1..key8, which could be deprecated on a longer timescale.

Contributions welcome !

**Comment by Spike-from-NH** [comment]
@Julow, good point, you could support nw= as well as key1= and you would not even have to deprecate or eventually remove the old format. I presume the inefficiency of checking for two alternatives is not massive nor something the app has to do continually.

Your point is valid that theme vs. key layout are separate. But, for instance, if I wanted a different look for my meta-keys (Shift, Alt, Ctrl, etc.), the layout file would specify which they are (with something like CLASS=) while the theme file specified how they look (with something like CSS). If you eventually supported style information in the layout, it would override the theme.

I agree a lot of new, not-strictly-keyboard features are "possible"—but "lightweight" is your chosen value and you should stick with it!
